### ISSUE TITLE
RowDecoder methods hidden from resolution once an instance is created 

### ISSUE THREAD

**Issue by ssimeonov** [nrinaudo/kantan.csv#34]
Attempting to create a decoder twice (or any other decoder for that matter) generates an error after one decoder is in scope.

``` scala
Welcome to Scala version 2.10.6 (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_66).
Type in expressions to have them evaluated.
Type :help for more information.

scala> case class Car(year: Int, make: String, model: String, desc: Option[String], price: Float)
defined class Car

scala> kantan.csv.RowDecoder.decoder5(Car.apply)(0, 1, 2, 3, 4)
res0: kantan.csv.RowDecoder[Car] = kantan.codecs.Decoder$$anon$1@3f39cd95

scala> kantan.csv.RowDecoder.decoder5(Car.apply)(0, 1, 2, 3, 4)
<console>:10: error: value decoder5 is not a member of object kantan.csv.RowDecoder
              kantan.csv.RowDecoder.decoder5(Car.apply)(0, 1, 2, 3, 4)
                                    ^
```

**Comment by nrinaudo** [comment]
I believe this is a REPL only issue - the Scala REPL has a long standing bug with objects that bear the same name as type aliases, which is exactly how RowDecoder works.

Did you encounter the problem in another context, and if so, would you be able to provide a working demonstration of the issue?

If we're only talking about the REPL thing, I don't consider it critical enough to rewire kantan.csv's innards. Let me know if it's a serious matter for you and I'll try to help you find a workaround.

**Comment by ssimeonov** [comment]
The issue is indeed a REPL one, but not in the usual sense. The reason why it's critical to me is that I'm using Spark (the most popular Scala-based open-source project and, I'd bet, the most common reason for anyone to use the Scala REPL right now). Spark sessions are essentially REPL sessions and are frequently used by non-developers. Further, Spark is mostly running on Scala 2.10.x so the default generic example (using shapeless) is not working either. Ideas for workarounds would be welcome.

**Comment by nrinaudo** [comment]
I *think* there should not be any issue with running the generic module under 2.10. [Here](http://scastie.org/16859)'s a scastie that shows it working.

If I had to guess, I'd say you lack the macroparadise plugin, it's a requirement for Shapeless on 2.10. I'll amend the documentation to state this.

**Comment by nrinaudo** [comment]
Did you manage to get it to work, and if so, is it enough of a workaround for your issue? If not, let me know and I'll see if I can come up with something.

**Comment by ssimeonov** [comment]
Alas, the REPL for the managed Spark service provider I am using (Databricks) doesn't seem to include the macroparadise plugin:

```scala
import scala.annotation.compileTimeOnly
<console>:33: error: object compileTimeOnly is not a member of package annotation
       import scala.annotation.compileTimeOnly
              ^
```

**Comment by nrinaudo** [comment]
!@#, if you'll pardon my French. This is going to be one of the tricky ones. I'll need to give this some thought. Do you think you could join the gitter channel so that we can discuss this live when we're both online?

**Comment by nrinaudo** [comment]
The Scala REPL bug is fixed for 2.12. I doubt the fix will be back-ported, but I don't really think there's more to be done here.
